GMG/Core/Models/ScoreFactory/ChordInferencer.swift

```diff
@@ -7,436 +7,547 @@ struct ChordInferencerResult {
+        let melodyArray: MLShapedArray<Int32> = MLShapedArray<Int32>(
+            scalars: tokenIds.map(Int32.init),
+            shape: [1, maxMelodyLength]
+        )
+
+        let melodyTokens = MLMultiArray(melodyArray)
+
+        let chordInferenceResults = try generateOutputSequence(
+            melodyTokens: melodyTokens,
+            previousResults: previousResults,
+            topK: 5,
+            temperature: 0.9
+        )
+
+        return chordInferenceResults
     }
 
     private func generateOutputSequence(
         melodyTokens: MLMultiArray,
+        previousResults: [[InferenceResult]]? = nil,
         topK: Int? = 5,
         temperature: Float = 0.9
-    ) throws -> [Int] {
-        var chordIndices: [Int] = [sosIndex]
-
-        // 모델 기대 타깃 길이로 배열 준비
-        let tgtLen = maxTgtLen
-        let chordInput = try MLMultiArray(
-            shape: [1, tgtLen] as [NSNumber],
-            dataType: .int32
+    ) throws -> [[InferenceResult]] {
+        var results: [[InferenceResult]] = previousResults ?? []
+
+        var chordArray: MLShapedArray<Int32> = MLShapedArray(
+            repeating: Int32(padId),
+            shape: [1, maxChordLength]
         )
 
-        for _ in 0..<tgtLen {
-            // 입력 채우기
-            for i in 0..<tgtLen {
-                let val = (i < chordIndices.count) ? chordIndices[i] : padIndex
-                chordInput[[0, i] as [NSNumber]] = NSNumber(value: val)
-            }
+        chordArray[scalarAt: [0, 0]] = Int32(sosId)
+        for (index, result) in results.enumerated() {
+            let tokenId: Int = result.first?.id ?? unkId
+            chordArray[scalarAt: [0, index + 1]] = Int32(tokenId)
+        }
 
-            let input = TransformerChordInferenceInput(
-                melody_tokens: melodyTokens,
-                chord_input: chordInput
+        for index in (results.count + 1)..<maxChordLength {
+            let chordInput: MLMultiArray = MLMultiArray(chordArray)
+
+            let input: TransformerChordInferenceInput =
+                TransformerChordInferenceInput(
+                    melody_tokens: melodyTokens,
+                    chord_input: chordInput
+                )
+            let output: TransformerChordInferenceOutput = try model.prediction(
+                input: input
             )
-            let output = try model.prediction(input: input)
-            let logits = output.chord_output
 
-            // step 위치의 다음 토큰 로짓 사용
-            let pos = chordIndices.count - 1
-            let nextToken = try sampleNextToken(
+            let logits: MLShapedArray<Float> = output.chord_outputShapedArray
+
+            let position: Int = results.count
+
+            let candidates: [InferenceResult] = try topKTokens(
                 from: logits,
-                at: pos,
+                at: position,
                 temperature: temperature,
-                topK: topK
+                k: 5
```

Comment:
해당 부분 위에 TopK 선언해놓은 것 같아서 재사용하면 될 것 같습니다!
```suggestion
                k: topK ?? 5
```